tools/codegen/api/unboxing.py

```diff
@@ -0,0 +1,201 @@
+from typing import List, Tuple
+
+from tools.codegen.api import cpp
+from tools.codegen.api.types import Binding, CType, CppSignatureGroup
+from tools.codegen.model import (
+    Argument,
+    NativeFunction,
+    Type,
+    BaseType,
+    OptionalType,
+    ListType,
+    BaseTy,
+)
+
+# This file generates the code for unboxing wrappers, i.e., the glue logic to unbox a boxed operator and convert the
+# ivalues from stack to correct arguments to the unboxed kernel, based on corresponding JIT schema. This codegen is
+# an alternative way to generate unboxing wrappers similar to the existing C++ metaprogramming approach but gets the
+# job done statically. These generated unboxing wrappers will be useful under the scenario where we need to register
+# a fixed set of operators known at compile time and thus can save some time in runtime initialization phase.
+#
+# Here's an example on how the codegen works:
+#
+# - Function Schema (source of truth)
+#
+#      aten::empty.names(int[] size, *, Dimname[]? names,
+#                        ScalarType? dtype=None, Layout? layout=None,
+#                        Device? device=None, bool? pin_memory=None,
+#                        MemoryFormat? memory_format=None) -> Tensor
+# - Argument Conversion
+#       Generates C++ code to convert an ivalue (from stack) to its underlying C++ type.
+#    - int[] size
+#        ```cpp
+#           const c10::List<c10::IValue> size_list_in = (std::move(peek(stack, 0, 7))).toList();
+#
+#           std::vector<int64_t> size_vec;
+#           for (c10::IValue size_elem: size_list_in) {
+#               int64_t size_base = size_elem.to<int64_t>();
+#               size_vec.push_back(size_base);
+#           }
```

Comment:
@ysiraichi we decided it wasn't doing the https://github.com/pytorch/pytorch/pull/67964 treatment for int lists, right?

Reply:
I don't think we discussed about int lists. But, we can add support for them with #69606 changes.

Reply:
I'm a little torn about this because int lists aren't refcounted so copying into a contiguous array isn't too bad; plus people are more likely to repeatedly index into int lists. On the other hand you may be able to avoid a dynamic allocation if the int list size is statically known.